umapi_client/functional.py

```diff
@@ -119,12 +133,12 @@ def __init__(self, id_type=IdentityTypes.adobeID, email=None, username=None, dom
             elif id_type is not IdentityTypes.federatedID:
                 raise ArgumentError("Username must match email except for Federated ID")
             else:
-                self._validate(username=username)
+                self._validate_username(username=username)
```

Comment:
This is not correct anymore, because we allow email-formatted usernames that don't match the user's email in Federated, but your validator doesn't allow emails.  See comment above.

Reply:
Please see my other two comments. We need separate validator functions for a few important reasons.

Reply:
I'm not objecting to the separate validator functions, which I think make perfect sense.  I'm just objecting to the username validator being overly restrictive for no good reason.